### PR TITLE
Clarify test name filter usage

### LINKED ISSUE #8282: Filter and build integration tests needed only
**Describe the problem you are trying to solve**

There is a project with 20+ integration tests, it takes like few minutes to build all of them. Even with a small change in one of the tests file and I wanted to just run one of the tests file, I did `cargo tests part_a` but it still compiles everything again.

The solution right now,

1. `rm tests/*`
2. `git restore tests/part_a.rs`
3. `cargo test part_a`, try rebuilding
4. `git restore src/dep.rs`, add some dependencies needed

Essentially, this will also remove the issue of clogging the screen with 0 tests run and the need to scroll up multiple times to search for the tests run #4324.

**Describe the solution you'd like**

Search for the functions used in the filter (`cargo test <filter>`), build and run only those files needed. Something like ripgrep for function name, but this requires cargo to include the dependencies too.

This provides 2 benefits, reducing the need to build unneeded files and prevent clogging the screen by running file without tests matching the filter.

**Notes**

I would like to take on this issue if the complexity is not high or if there are people willing to mentor.

### PR DESCRIPTION
**Pull request by icorbrey**
I set aside the description of the usage example in the top description section and added a more concise example in the example section at the bottom of the man page. I also changed the wording of the usage example up top a bit to be more search-friendly since "filter" is a common keyword to search for.

Resolves #8282

### CODE REVIEW

**Comment by alexcrichton** [comment]
Looks reasonable to me, thanks! We currently check in generated versions of the documentation as well, so I think this'll need to run `make` inside of `src/doc` to generate the HTML and such.

**Comment by icorbrey** [comment]
Ah, I forgot about the generated docs. Give me a minute to get those compiled lol

File: src/etc/man/cargo-test.1 [hunk 1]
```diff
@@ -1,66 +1,63 @@
 '\" t
 .\"     Title: cargo-test
-.\"    Author: [see the "AUTHOR(S)" section]
-.\" Generator: Asciidoctor 2.0.10
-.\"      Date: 2020-06-25
+.\"    Author: [see the "AUTHORS" section]
+.\" Generator: Asciidoctor 1.5.5
+.\"      Date: 2020-07-28
```
Comment (icorbrey) [inline src/etc/man/cargo-test.1:5]: The manpage got a lot of weird little changes like this, was this caused by my Asciidoctor version? This was just the default version that I installed from `apt`, so I'm not sure if that's fine or not. Plus there's a ton of stuff replacing `\fB` with `\f[CR]` (I assume some kind of "line break -> carriage return" thing).

Comment (icorbrey) [reply src/etc/man/cargo-test.1:5]: Not like the date stuff, that makes sense

**Comment by ehuss** [comment]
Hm, yea, strange, possibly a different version.  I just pushed an update to rebuild the affected page.  I have a new tool that will change how this is built, hopefully that will land soonish to avoid issues like this.

Thanks again!

@bors r+

**Comment by icorbrey** [comment]
No problem, hopefully there's more issues that I can help get resolved. I'm just getting started in Rust but it has a ton of potential and I want to be able to give back to the community 😁

**Pull request merged**